File: nft-recsys-proxy/controllers/item-info.go
package controllers

import (
	"github.com/gin-gonic/gin"
	"net/http"
	"nft-recsys-proxy/util"
)

// GetInfoTrait  Get item info for trait-based rec items
func GetInfoTrait(c *gin.Context) {
	referenceId := c.Query("reference_id")

	url := util.TraitSimilarityRecSysEndpoint + "/ref-info?reference_id=" + referenceId
	remoteResponse := util.GetFromRemoteAPI(url, c)

	if remoteResponse == nil {
		return
	}

	c.JSON(http.StatusOK, gin.H{
		"status": "success",
		"data":   remoteResponse,
	})
}

// GetInfoBasicContent  Get item info for basic content-based rec items
func GetInfoBasicContent(c *gin.Context) {
	referenceId := c.Query("reference_id")

	url := util.BasicContentRecSysEndpoint + "/ref-info?reference_id=" + referenceId
	remoteResponse := util.GetFromRemoteAPI(url, c)

	if remoteResponse == nil {
		return
	}

	c.JSON(http.StatusOK, gin.H{
		"status": "success",
		"data":   remoteResponse,
	})
}

File: nft-recsys-frontend/src/services/item-info.js
import axios from "axios";
import { refItemInfoEndpoint } from "../endpoints";

// Get reference item info - basic content
export async function retrieveItemInfoBasicContent(initRefItem) {
    const res = await axios.get(
        refItemInfoEndpoint + `/basic-content?reference_id=${initRefItem}`,
        {}
    );

    return res.data;
}

// Get reference item info - trait
export async function retrieveItemInfoTrait(initRefItem) {
    const res = await axios.get(
        refItemInfoEndpoint + `/trait?reference_id=${initRefItem}`,
        {}
    );

    return res.data;
}
